src/manual_inspection.py

```diff
@@ -0,0 +1,183 @@
+
+    def _get_labeled_images(self):
+        """Retrieve a set of already labeled images from the CSV file."""
+        if self.csv_file.exists():
+            df_existing = pd.read_csv(self.csv_file)
+            return set(df_existing['ImageID'].tolist())
+        return set()
+
+    def _load_existing_results(self):
+        """Load existing CSV results or return an empty list."""
+        if self.csv_file.exists():
+            log.info(f"Loading existing results from {self.csv_file}")
+            return pd.read_csv(self.csv_file).values.tolist()
+        return []
+
+    def _display_instructions(self):
+        """Prints instructions for user input."""
+        print("\n--- Segmentation Quality Assessment ---")
+        print_labels = {
+            "1": "Good Mask",
+            "2": "Bad Mask",
+            "3": "Incorrect Species",
+            "0": "Other",
+            "q": "Quit"
+        }
+        
+        for key, label in print_labels.items():
+            if key == "0":
+                print(f"{key}️ (zero) - {label}")
+            else:
+                print(f"{key}️ - {label}")
+        print("\n🔄 Please wait while the X11 or X410 forwarding initializes. This may take a few seconds...\n")
+
+    def _display_image(self, img_path):
+        """Loads and displays an image, returns False if loading fails."""
+        image = cv2.imread(str(img_path))
+        if image is None:
+            log.error(f"⚠️ Error loading image: {img_path}")
+            return False
+        
+        cv2.imshow("Inspection Viewer", image)
+        return True
+
+    def _get_user_input(self):
+        """Captures user input for labeling images."""
+        while True:
+            key = cv2.waitKey(0) & 0xFF
+            key_char = chr(key)
+            if key_char in LABEL_OPTIONS:
+                return LABEL_OPTIONS[key_char]
+            print("⚠️ Invalid choice. Please press a number between 0-3 or 'q' to quit.")
+
+    def _save_results(self):
+        """Save the labeling results to a CSV file."""
+        df = pd.DataFrame(self.results, columns=['BatchID', 'ImageID', 'Selection', 'Timestamp', 'User', 'LTSLocation'])
+        df.to_csv(self.csv_file, index=False)
+
+    def _review_flagged_images(self):
+        """Checks and offers to display flagged images for issue reporting."""
+        df_final = pd.read_csv(self.csv_file)
+        flagged_images = df_final[df_final["Selection"] != "Good Mask"]
+
+        if flagged_images.empty:
+            return self.csv_file
+
+        print("\n⚠️ Some images have issues.")
+        print(f"📌 Please report in our GitHub repository: {GITHUB_REPO_URL}")
+        print("Mention the flagged images and describe the issues.")
+        if input("Would you like to review the flagged images for screenshots? (y/n): ").strip().lower() == 'y':
+            print("Waiting to allow the image viewer to load...")
+            time.sleep(10)
```

Comment:
Not sure what this is for. You're delaying the entire script no?

Reply:
I observed that if i press 'y' and enter too quick, the image viewer crashes. That's why i added it. Others might find a similar issue. It is only one time '10' second delay.

Reply:
How are you using x11 forwarding?

In another separate terminal (not vs code) ssh into sunny using

ssh -X -C <username>@SUNNY.ece.ncsu

Reply:
I am using a separate terminal and ssh -X  ___ @SUNNY.ece.ncsu